doc/ug/dv_methodology.md

```diff
@@ -299,65 +299,124 @@ Once the regression tool is developed, it will provide a way to enable these cap
-
-These constructs are typically encapsulated in a class and are sometimes referred to as 'functional coverage model'.
-They are sampled in 'reactive' components of the testbench, such as monitors and / or the scoreboards.
+These are more complex constructs that capture whether signals (that reflect the current state of the design) have met a set of values or a sequence of values that are interesting / corner cases.
+These constructs also allow us to capture whether multiple scenarios have occurred simultaneously through crosses.
+This is also often referred to as cross coverage.
+These constructs are typically encapsulated in a class and are sometimes referred to as the ‘functional coverage model’.
+They are sampled in ‘reactive’ components of the testbench, such as monitors and/or the scoreboards.
 They can also be embedded within the RTL to sample interesting scenarios through DV stimulus.
 
 Here are the metrics used with a brief explanation:
 
-*  **Assert Coverage**: This metric measures which assertions, cover properties and sequences have been exercised in the course of the simulation.
+* **Assert Coverage**: This metric measures which assertions, cover properties, and sequences have been exercised in the course of the simulation.
   Note, an assertion precondition counts as a cover point.
-*  **Covergroup Coverage**: This metric (sometimes called "Testbench Coverage") measures covergroups during simulation.
-  These are usually coded in the testbench to check that the desired stimulus was generated properly, but can also be embedded in the RTL.
-
-Code coverage can reach 100% but it may not be sufficient to indicate whether all interesting combinations of scenarios were exercised (such as boundary conditions, concurrent scenarios and different CSR fields holding specific combinations of values resulting in the DUT being exercised in interesting ways).
-Hence, developing a detailed functional coverage model in conjunction with constrained random test sequences is key to high quality DV effort.
-
-Functional coverage can also reach 100% while the code coverage may not.
-This would indicate that the testplan is either incomplete or there are undocumented or unreachable features in the design.
-As such they can be addressed accordingly through either updating the design specification and augmenting the testplan / written tests, or by optimizing the design to remove unreachable logic if possible.
-There may be features that cannot be tested and cannot be removed from the design either.
-These would have to be analyzed and excluded from the coverage collection as a waiver.
-This is an exercise the DV and the designer typically perform together.
-This is discussed in more detail below.
+* **Cover group Coverage**:  Observes values on busses, registers, etc.
+  can verify that a specific value or a range of values was seen on a bus.
```

Comment:
This is not limited to design elements such as bus and registers - you could write covergroups that look at values / combinations of values at higher levels (functional) of abstraction too (example: saw a shadow register storage error while the AES was busy).

Reply:
Not sure I follow this @sriyerg 
this is meant as an example, not an exhaustive list. 
if this is interpreted as a "limited to" then I need to reword.

Anyway the higher level of abstraction you are talking about I believe is covered under my small comment on cross-coverage which follows on line 363?

I am open to more general suggestions if you feel my text is to limited.

Reply:
The initial description of the bullet should provide a general explanation of what it means to write covergroups (and what kinds of scenarios they capture). It can then be followed up with specific examples. 
IMO, the description itself should not be a narrow / limited view of covergroups.

Reply:
so what If I add 

> as an example this can be a value or a range of values seen on a bus